### PR TITLE
Simplify cronjob v2 controller tests

### PR DESCRIPTION
**Pull request by soltysh**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
This simplifies and unifies cronjob v2 controller unit tests. 

#### Special notes for your reviewer:
/assign @damemi 

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/hold 
until the rest of gang merges

### CODE REVIEW

File: pkg/controller/cronjob/cronjob_controllerv2_test.go [hunk 1]
```diff
@@ -322,230 +323,156 @@ func (f *fakeQueue) AddAfter(key interface{}, t time.Duration) {
...
-		newJobSchedule string
-	}
+func TestControllerV2UpdateCronJob(t *testing.T) {
 	tests := []struct {
 		name              string
-		fields            fields
-		args              args
+		oldCronJob        *batchv1beta1.CronJob
+		newCronJob        *batchv1beta1.CronJob
 		deltaTimeForQueue time.Duration
```
Comment (alaypatel07) [inline pkg/controller/cronjob/cronjob_controllerv2_test.go:331]: Maybe this can be named to `expectedDeltaTimeForRequeue` for readibility?

**Review by alaypatel07 — COMMENTED** [review]
added a question, apart from that this looks great, thanks @soltysh 

/lgtm

File: pkg/controller/cronjob/cronjob_controllerv2_test.go (hunk 1 above)
Comment (soltysh) [reply pkg/controller/cronjob/cronjob_controllerv2_test.go:331]: Good point will update once the v1 bits are in.

**Review by damemi — COMMENTED** [review]
/lgtm

**Pull request merged**